fastlane/lib/fastlane/actions/upload_symbols_to_crashlytics.rb

```diff
@@ -2,9 +2,64 @@ module Fastlane
   module Actions
     class UploadSymbolsToCrashlyticsAction < Action
       def self.run(params)
-        params[:binary_path] ||= (Dir["/Applications/Fabric.app/**/upload-symbols"] + Dir["./Pods/**/upload-symbols"]).last
-        UI.user_error!("Please provide a path to the binary using binary_path:") unless params[:binary_path]
+        require 'tmpdir'
+
+        find_binary_path(params)
+        find_api_token(params)
+
+        dsym_paths = []
+        dsym_paths << params[:dsym_path]
+        dsym_paths += Actions.lane_context[SharedValues::DSYM_PATHS] if Actions.lane_context[SharedValues::DSYM_PATHS]
+        UI.error("Couldn't find any dSYMs, please pass them using the dsym_path option") if dsym_paths.count == 0
```

Comment:
That use case is interesting and certainly makes life easier for the user who doesn't have to check if there are dsyms before calling upload. Let's return after this if no dsyms.